packages/object/addon/index.js

```diff
@@ -188,17 +194,21 @@ export const observes = decoratorWithRequiredParams((desc, params) => {
   @function
   @param {...String} propertyNames - Names of the properties that no longer trigger the function
  */
-export const unobserves = decoratorWithRequiredParams((desc, params) => {
-  desc.finisher = target => {
-    let { prototype } = target;
-
-    for (let path of params) {
-      removeObserver(prototype, path, null, desc.key);
-    }
+export const unobserves = decoratorWithRequiredParams(
+  (desc, params) => ({
+    ...desc,
+    finisher(target) {
```

Comment:
Do we prefer this idiom over

```js
  (desc, params) => ({
    ...desc,
    finisher(target) {
      // ...
    }
  })
```

or

```js
  (desc, params) => {
    return {
      ...desc,
      finisher(target) {
        // ...
      }
    }
  }
```

Or is the `...spread` operator potentially creating bloat in transpiled code? Then a mix of directly mutating the desc and returning it might be preferable.

Reply:
I don't have a strong preference style-wise here, I think `return` style is a little more readable at first, but once you internalize `({})` style it's not that much different and much less verbose so that's nice.

I think the spread operator won't add much in terms of bloat once we get externalized helpers working, but we are creating a new object instead of changing the original one, unsure how that could affect things perf-wise but in hot paths I know it can matter.

Reply:
I would expect that at least for evergreen browsers the `{ ...old, override: foo }` pattern will be quite performant, because it is used massively by Redux users.

But since there is no _actual need_ to create a new object, mutating and returning the existing one is _probably_ a bit faster and just as good, though not as idiomatic.

I also don't mind either way. Maybe someone else would like to weigh in?

Reply:
I've updated the code to use the mutate and return pattern, because it introduces less noise to this PR and is potentially a bit faster and means no code bloat even without externalized helpers.